zookeeper-server/src/main/java/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -1510,6 +1511,11 @@ public void close() throws IOException {
      * the server. Thus, getXid() must be public.
      */
     synchronized public int getXid() {
+        // xid values of -4, -2, and -1 are special, see SendThread.readResponse
+        // Skip from -5 to 1.
```

Comment:
why are we allowing negative numbers at all? shouldn't we check for "about to rollover" prior to the increment and then reset to 1? It seems to me that that would be a whole lot easier to grok.

Reply:
A reasonable point.  Nothing I could find directly compares cxid values or will otherwise notice the MAX to MIN wrap, so I chose to use the full range.  I'll switch it.

Reply:
Sounds good, thanks. Essentially negative cxid are essentially "reserved" this way.